Lib/_pyio.py

```diff
@@ -617,6 +617,8 @@ def read(self, size=-1):
         n = self.readinto(b)
         if n is None:
             return None
+        if n < 0 or n > len(b):
+            raise ValueError(f"readinto returned '{n}' outside buffer size '{len(b)}'")
```

Comment:
I would prefer to not use quotes around integers:

```suggestion
            raise ValueError(f"readinto returned {n} outside buffer size {len(b)}")
```

Same remark for the C code.